### ISSUE TITLE
Having GitUp play nicely with 1Password 8's ssh features

### ISSUE THREAD

**Issue by johncoxon** [git-up/GitUp#810]
I get the error "Failed connecting to "origin" remote: authentication required but no callback set" (the same as in #35) with <a href="https://blog.1password.com/1password-ssh-agent/">1Password 8's new ssh feature for git repos</a>. Pushing from the Terminal works as expected – how can I push from GitUp?

**Comment by lucasderraugh** [comment]
#35 is a solved problem for all intents and purposes (adding `ssh-add --apple-use-keychain -q` to your `.zshrc` file and opening a terminal window). You can also load them on startup which I haven't done personally but it should work in theory: https://www.getpagespeed.com/work/proper-use-of-ssh-client-in-mac-os-x

That said, I'll have to research it a bit more to understand what they're doing that is special from the usual path. Thanks for bringing this up though.

**Comment by johncoxon** [comment]
<a href="https://developer.1password.com/docs/ssh/agent/compatibility/#ssh-auth-sock">If you use this snippet from 1Password it fixes this issue.</a>

**Comment by lucasderraugh** [comment]
Thanks @johncoxon for the research and the update!

**Comment by haydn** [comment]
I've got a fresh macOS install and I don't seem to be able to get this working. I've used the `SSH_AUTH_SOCK` setup as described in 1Password's docs (added a `com.1password.SSH_AUTH_SOCK.plist` file to `~/Library/LaunchAgents`) and it now prompts me to authenticate when I interact with a remote in GitUp, but then it's immediately followed by that same error message:

![image](https://user-images.githubusercontent.com/27211/170398055-d7a17df4-eb80-44a6-b99b-a76212450bcd.png)

I'm able to successfully `git fetch` via the terminal using 1Passwords prompt, so it's something specific to the GitUp setup. Anything obvious I might've setup wrong?
